src/library/scala/collection/SortedMap.scala

```diff
@@ -122,7 +122,7 @@ trait SortedMapOps[K, +V, +CC[X, Y] <: Map[X, Y] with SortedMapOps[X, Y, CC, _],
 
   /** The implementation class of the set returned by `keySet` */
   protected class KeySortedSet extends SortedSet[K] with GenKeySet with GenKeySortedSet {
-    def diff(that: Set[K]): SortedSet[K] = fromSpecificIterable(view.filterNot(that))
+    def diff(that: IterableOnce[K]): SortedSet[K] = SortedSet.from(this).diff(that)
```

Comment:
The previous implementation seems cheaper (no intermediate `SortedSet.from(this)` construction). Why wasn’t it still working?

Reply:
`that` is not a `Set` anymore, so `filterNot` would be very inefficient. Another implementation could be `fromSpecificIterable(view.filterNot(that.toSet))` but you do create an intermediate collection.